docs_src/microservices/application/ContextAPI.md

```diff
@@ -1,108 +1,142 @@
-	Configuration common.ConfigurationStruct
-	
-	// LoggingClient is exposed to allow logging following the preferred logging strategy within EdgeX.
-	LoggingClient logger.LoggingClient
-	
-	// EventClient exposes Core Data's EventClient API
-	EventClient coredata.EventClient
-	
-	// ValueDescriptorClient exposes Core Data's ValueDescriptor API
-	ValueDescriptorClient coredata.ValueDescriptorClient
-	
-	// CommandClient exposes Core Commands's Command API
-	CommandClient command.CommandClient
-	
-	// NotificationsClient exposes Support Notification's Notifications API
-	NotificationsClient notifications.NotificationsClient
-	
-	// RetryData holds the data to be stored for later retry when the pipeline function returns an error
-	RetryData []byte
-	
-	// SecretProvider exposes the support for getting and storing secrets
-	SecretProvider *security.SecretProvider
-
-	// ResponseContentType sets a custom response type
-	ResponseContentType string
+type AppFunctionContext interface {
+    CorrelationID() string
+    InputContentType() string
+    SetResponseData(data []byte)
+    ResponseData() []byte
+    SetResponseContentType(string)
+    ResponseContentType() string
+    SetRetryData(data []byte)
+    GetSecret(path string, keys ...string) (map[string]string, error)
+    SecretsLastUpdated() time.Time
+    LoggingClient() logger.LoggingClient
+    EventClient() interfaces.EventClient
+    CommandClient() interfaces.CommandClient
+    NotificationClient() interfaces.NotificationClient
+    SubscriptionClient() interfaces.SubscriptionClient
+    DeviceServiceClient() interfaces.DeviceServiceClient
+    DeviceProfileClient() interfaces.DeviceProfileClient
+    DeviceClient() interfaces.DeviceClient
+    PushToCore(event dtos.Event) (common.BaseWithIdResponse, error)
+    GetDeviceResource(profileName string, resourceName string) (dtos.DeviceResource, error)
+    AddValue(key string, value string)
+    RemoveValue(key string)
+    GetValue(key string) (string, bool)
+    GetAllValues() map[string]string
+    ApplyValues(format string) (string, error)
 }
 ```
 
+## InputContentType()
+`InputContentType()` returns the content type of the data that initiated the pipeline execution. Only useful when the TargetType for the pipeline is []byte, otherwise the data with be the type specified by TargetType.
+
+## SetResponseData()
+`SetResponseData(data []byte)` sets the response data that will be returned to the trigger when pipeline execution is complete.
+
+## ResponseData()
+`ResponseData()` returns the data that will be returned to the trigger when pipeline execution is complete.
+
+## SetResponseContentType()
+`SetResponseContentType(string)` sets the content type that will be returned to the trigger when pipeline execution is complete.
+
+## ResponseContentType()
+`ResponseContentType()` returns the content type that will be returned to the trigger when pipeline execution is complete.
+
+## SetRetryData()
```

Comment:
This is duplicated below. Remove below version, but retain note from below.

Reply:
Also create `Miscellaneous` header at bottom and put this under it with other noted below.